### ISSUE TITLE
Non-ascii code is not supported

### ISSUE THREAD

**Issue by serhiy-storchaka** [haypo/perf#15]
```
$ ./python -m perf timeit -s 's="АБВГҐДЕЄЖЗИІЇЙКЛМНОПРСТУФХЦЧШЩЬЮЯ"' 's.count("Є")'
Traceback (most recent call last):
  File "/home/serhiy/py/cpython/Lib/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/serhiy/py/cpython/Lib/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/__main__.py", line 727, in <module>
    main()
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/__main__.py", line 719, in main
    func()
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/__main__.py", line 512, in cmd_timeit
    perf._timeit.main(timeit_runner)
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/_timeit.py", line 184, in main
    timer = create_timer(runner, stmt)
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/_timeit.py", line 104, in create_timer
    return timeit.Timer(stmt, setup, timer=perf.perf_counter)
  File "/home/serhiy/py/cpython/Lib/timeit.py", line 109, in __init__
    compile(setup, dummy_src_name, "exec")
UnicodeEncodeError: 'utf-8' codec can't encode characters in position 3-68: surrogates not allowed
Traceback (most recent call last):
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/_timeit.py", line 194, in main
    runner.bench_sample_func(args.name, sample_func, timer, **kwargs)
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/_runner.py", line 529, in bench_sample_func
    return self._main(name, wrap_sample_func, inner_loops)
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/_runner.py", line 496, in _main
    bench = self._master()
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/_runner.py", line 732, in _master
    bench = self._spawn_workers()
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/_runner.py", line 689, in _spawn_workers
    suite = self._spawn_worker(calibrate)
  File "/home/serhiy/.local/lib/python3.7/site-packages/perf/_runner.py", line 626, in _spawn_worker
    % (cmd[0], exitcode))
RuntimeError: /home/serhiy/py/cpython/python failed with exit code 1
```

Standard `timeit` module supports non-ascii code.
```
$ ./python -m timeit -s 's="АБВГҐДЕЄЖЗИІЇЙКЛМНОПРСТУФХЦЧШЩЬЮЯ"' 's.count("Є")'
500000 loops, best of 5: 410 nsec per loop
```

**Comment by haypo** [comment]
When perf spawns a worker process, it creates a new empty environment (env vars): locale variables are removed. The idea is to get more reproductible benchmarks.

I didn't think about the encoding of the Linux command line.

You can workaround the issue by inheriting explicitly locale env vars: add --inherit=LC_ALL,LANG,LC_CTYPE to your command line.

It's not convenient, something should be fixed.

**Comment by haypo** [comment]
The issue should be fixed by 5fbc4fbc76f391639d94830ecc9126d8ddfb3354: can you please try the development version of perf?

**Comment by serhiy-storchaka** [comment]
The issue is fixed with the current version of perf. Thanks!

**Comment by haypo** [comment]
Oops, sorry, I fixed the bug when I saw the bug report but I forgot to
update the issue.

Since the start of perf, I'm trying to make benchmarks more stable and more
reproductible. I decided to remove all env vars since it's common that an
user has a few specific vars for different reasons and I didn't want to get
different perfs because of that. I prefer to have a white list of env var
to help users to understand the impact of vars on perf.

*But* this issue is a clear example showing that my heuristic wasn't
perfect. The difference between regular Python and perf was too important
and the behaviour was too surprising.

So perf now copies locale vars and I added --no-locale option to explicitly
not copy the locale. I inverted the default behaviour.

Thanks for your bug report Serhiy. My TODO list is now empty, next release
is likely to be perf 1.0.
